server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailSetMethod.scala

```diff
@@ -0,0 +1,225 @@
+    for {
+      destroyResults <- destroy(request, mailboxSession)
+      updateResults <- update(request, mailboxSession)
+    } yield InvocationWithContext(
+        invocation = Invocation(
+          methodName = invocation.invocation.methodName,
+          arguments = Arguments(serializer.serialize(EmailSetResponse(
+            accountId = request.accountId,
+            newState = State.INSTANCE,
+            updated = updateResults.updated,
+            destroyed = destroyResults.destroyed,
+            notDestroyed = destroyResults.notDestroyed))),
+          methodCallId = invocation.invocation.methodCallId),
+        processingContext = invocation.processingContext)
+  }
+
+  override def getRequest(mailboxSession: MailboxSession, invocation: Invocation): SMono[EmailSetRequest] = asEmailSetRequest(invocation.arguments)
+
+  private def asEmailSetRequest(arguments: Arguments): SMono[EmailSetRequest] =
+    serializer.deserialize(arguments.value) match {
+      case JsSuccess(emailSetRequest, _) => SMono.just(emailSetRequest)
+      case errors: JsError => SMono.raiseError(new IllegalArgumentException(ResponseSerializer.serialize(errors).toString))
+    }
+
+  private def destroy(emailSetRequest: EmailSetRequest, mailboxSession: MailboxSession): SMono[DestroyResults] =
+    SFlux.fromIterable(emailSetRequest.destroy.getOrElse(DestroyIds(Seq())).value)
+      .flatMap(id => deleteMessage(id, mailboxSession))
+      .collectSeq()
+      .map(DestroyResults)
+
+  private def update(emailSetRequest: EmailSetRequest, mailboxSession: MailboxSession): SMono[UpdateResults] = {
+    emailSetRequest.update
+      .filter(_.nonEmpty)
+      .map(update(_, mailboxSession))
+      .getOrElse(SMono.just(UpdateResults(Seq())))
+  }
+
+  private def update(updates: Map[UnparsedMessageId, EmailSetUpdate], session: MailboxSession): SMono[UpdateResults] = {
+    val validatedUpdates: List[Either[UpdateFailure, (MessageId, EmailSetUpdate)]] = updates
+      .map({
+        case (unparsedMessageId, updatePatch) => EmailSet.parse(messageIdFactory)(unparsedMessageId)
+          .map((_, updatePatch))
+          .toEither
+          .left.map(e => UpdateFailure(unparsedMessageId, e))
+      })
+      .toList
+    val failures: List[UpdateFailure] = validatedUpdates.flatMap({
+      case Left(e) => Some(e)
+      case _ => None
+    })
+    val validUpdates: List[(MessageId, EmailSetUpdate)] = validatedUpdates.flatMap({
+      case Right(pair) => Some(pair)
+      case _ => None
+    })
+
+    for {
+      updates <- SFlux.fromPublisher(messageIdManager.messagesMetadata(validUpdates.map(_._1).asJavaCollection, session))
+        .collectMultimap(metaData => metaData.getComposedMessageId.getMessageId)
+        .flatMap(metaData => {
+          SFlux.fromIterable(validUpdates)
+            .flatMap[UpdateResult]({
```

Comment:
Use concatMap here to not create contention on the modseq generation (monotic integer generation relying on optimistic concurrency)